lib/client/index.js

```diff
@@ -0,0 +1,1710 @@
+
+    yAxis2 = d3.svg.axis()
+      .scale(yScale2)
+      .tickFormat(d3.format('d'))
+      .tickValues(tickValues)
+      .orient('right');
+
+    // setup a brush
+    brush = d3.svg.brush()
+      .x(xScale2)
+      .on('brushstart', brushStarted)
+      .on('brush', brushed)
+      .on('brushend', brushEnded);
+
+    updateChart(true);
+  }
+
+// get the desired opacity for context chart based on the brush extent
+  function highlightBrushPoints(data) {
+    if (data.mills >= brush.extent()[0].getTime() && data.mills <= brush.extent()[1].getTime()) {
+      return futureOpacity(data.mills - latestSGV.mills);
+    } else {
+      return 0.5;
+    }
+  }
+
+  function addPlaceholderPoints () {
+    // TODO: This is a kludge to advance the time as data becomes stale by making old predictor clear (using color = 'none')
+    // This shouldn't need to be generated and can be fixed by using xScale.domain([x0,x1]) function with
+    // 2 days before now as x0 and 30 minutes from now for x1 for context plot, but this will be
+    // required to happen when 'now' event is sent from websocket.js every minute.  When fixed,
+    // remove this code and all references to `type: 'server-forecast'`
+    var last = _.last(data);
+    var lastTime = last && last.mills;
+    if (!lastTime) {
+      console.error('Bad Data, last point has no mills', last);
+      lastTime = Date.now();
+    }
+
+    var n = Math.ceil(12 * (1 / 2 + (now - lastTime) / SIXTY_MINS_IN_MS)) + 1;
+    for (var i = 1; i <= n; i++) {
+      data.push({
+        mills: lastTime + (i * FIVE_MINS_IN_MS), mgdl: 100, color: 'none', type: 'server-forecast'
+      });
+    }
+  }
+
+// clears the current user brush and resets to the current real time data
+  function updateBrushToNow(skipBrushing) {
+
+    // get current time range
+    var dataRange = d3.extent(data, dateFn);
+
+    // update brush and focus chart with recent data
+    d3.select('.brush')
+      .transition()
+      .duration(UPDATE_TRANS_MS)
+      .call(brush.extent([new Date(dataRange[1].getTime() - foucusRangeMS), dataRange[1]]));
+
+    addPlaceholderPoints();
+
+    if (!skipBrushing) {
+      brushed(true);
+
+      // clear user brush tracking
+      brushInProgress = false;
+    }
+  }
+
+  function brushStarted() {
+    // update the opacity of the context data points to brush extent
+    context.selectAll('circle')
+      .data(data)
+      .style('opacity', 1);
+  }
+
+  function brushEnded() {
+    // update the opacity of the context data points to brush extent
+    context.selectAll('circle')
+      .data(data)
+      .style('opacity', function (d) { return highlightBrushPoints(d) });
+  }
+
+  function alarmingNow() {
+    return container.hasClass('alarming');
+  }
+
+  function inRetroMode() {
+    if (!brush) {
+      return false;
+    }
+
+    var time = brush.extent()[1].getTime();
+
+    return !alarmingNow() && time - TWENTY_FIVE_MINS_IN_MS < now;
+  }
+
+  function brushed(skipTimer) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/799312596/issues/source?bid=2452379&fileBranchId=2725007#l310)